pool/app/email/email_service.ml

```diff
@@ -4,17 +4,21 @@ module Queue = Sihl_queue.MariaDb
 module Cache = struct
   open Hashtbl
 
-  let tbl : (SmtpAuth.Id.t, SmtpAuth.Write.t) t = create 5
+  let tbl : (SmtpAuth.Id.t, Pool_database.Label.t * SmtpAuth.Write.t) t =
+    create 5
+  ;;
+
   let find_by_id id = find_opt tbl id
 
-  let find_default () =
+  let find_default pool =
     tbl
     |> to_seq_values
-    |> Seq.find (fun { SmtpAuth.Write.default; _ } ->
-      SmtpAuth.Default.value default)
+    |> Seq.find (fun (database_label, { SmtpAuth.Write.default; _ }) ->
+      SmtpAuth.Default.value default
+      && Pool_database.Label.equal pool database_label)
   ;;
 
-  let add ({ SmtpAuth.Write.id; _ } as m) = replace tbl id m
+  let add ((_, { SmtpAuth.Write.id; _ }) as m) = replace tbl id m
```

Comment:
What happens when two smtp configurations exist on different tenants with the same ID?

Is it possible to use the database label also within the identifier?